param/__init__.py

```diff
@@ -282,6 +288,27 @@ def _get_min_max_value(min, max, value=None, step=None):
     return min, max, value
 
 
+def _deserialize_from_path(ext_to_routine, path, type_name):
+    """Call deserialization routine with path according to extension"""
+    if not os.path.isfile(path):
+        raise FileNotFoundError("'{}' does not exist or is not a file".format(path))
+    root, ext = os.path.splitext(path)
+    if ext in {'.gz', '.bz2', '.xz', '.zip'}:
+        # A compressed type. We'll assume the routines can handle such extensions
+        # transparently
+        ext = os.path.splitext(root)[1]
+    # If the extension does match, this is likely the correct routine and we will error
+    # later on. Let's warn the user.
+    try:
+        if ext in ext_to_routine:
+            return ext_to_routine[ext](path)
+    except:
+        get_logger().warning(
+            "Could not parse file '{}' as {}".format(path, type_name), exc_info=True)
```

Comment:
One thing I think would be useful would be to state the supported extensions. Something like:

```python
"Could not parse file '{}' as {}. Supported extensions: {}".format(path, type_name, ', '.join(supported_exts))
```

The supported extensions would then probably be in some global registry by `type_name`.

Reply:
This specific warning occurs when the user tried to parse an extension we do support, but the routine failed to do so (e.g. the file was invalid). How about just below?